### DISCUSSION TITLE
Mutiple U Devices

### DISCUSSION THREAD

**Discussion by JHarrison712** [netbox-community/netbox#12287] [General]
We have several rack shelves that can support like an cable modem, maybe 5 port switch in the same U layout, can we get support on maybe allow 2 or 4 devices in the same U placement or say rack shelve - that can support layout?

Let me know if I need to take some screenshot of what I'm trying to say.

**Comment by candlerb** [comment]
You have two options with Netbox today:

1. Put the vanilla shelf in as a device, and put the cable modem and switch as "unracked" (i.e. rack is selected, but not rack position). Use the comments field to describe where the device is located.
2. Create the shelf as a "parent device", with "device bays", and make the cable modem and switch be "child devices" which are "inserted" into that shelf.

Personally I do (1), as I find(2) too icky.  Also, I don't think child devices are displayed at all in the rack detail view (you have to drill down into the parent device), whereas unracked devices are listed in a table.

**Reply by zmaster7** [reply]
I can confirm that child devices do not show up under "non-racked devices" when the parent device is in the rack, however you do get an indicator of device bays used vs total device bays on the parent device. With that said, I would go for option (1) as well, considering the devices on a shelf aren't always permanent.
